### PR TITLE
bpo-37284: Add note to sys.implementation doc

### PR DESCRIPTION
**Pull request by potomak**
Add a brief note to indicate that any new required attributes must go through the PEP process.

https://bugs.python.org/issue37284

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Our records indicate we have not received your CLA. For legal reasons we need you to sign this before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.

If you have recently signed the CLA, please wait at least one business day
before our records are updated.


You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for your contribution, we look forward to reviewing it!

**Review by aeros — CHANGES_REQUESTED** [review]
@potomak Thanks for the contribution! If you've already signed the CLA and it has been over 24 hours, try manually refreshing on status on GitHub (click on the "check yourself" link posted by the CLA bot @the-knights-who-say-ni). For mine, it was not automatically updated.

Usually when the tests are failing on documentation related PRs, it is a result of whitespace issues. You can attempt to correct this through GitHub's text editor, but the more reliable method is to locally clone the cpython repository using ``git clone``, make your changes to ``Doc/library/sys.rst`` (or download ``Doc/library/sys.rst`` from your ``patch-1`` branch into your local repo), use ``git add`` to add it to the checkout list, and then use the ``patchcheck`` utility (varies based on OS, let me know if you have further questions) to automatically correct whitespace issues. More details on this process can be found on the dev guide: https://devguide.python.org/pullrequest/?highlight=patchcheck#patchcheck

**Comment by potomak** [comment]
Thanks @aeros167 for your help.

I checked myself for the SLA signature and I updated the commit after running `make patchcheck` as you suggested.

**Comment by nanjekyejoannah** [comment]
cc @ericsnowcurrently

File: Doc/library/sys.rst [hunk 1]
```diff
@@ -880,6 +880,10 @@ always available.
 
    .. versionadded:: 3.3
 
+   .. note::
+
+      All proposals for new required attributes must go through the normal PEP
```
Comment (ericsnowcurrently) [inline Doc/library/sys.rst]: A slight tweak to make this more clear and direct:

```suggestion
      The addition of new required attributes must go through the normal PEP
```

**Review by ericsnowcurrently — APPROVED** [review]
Thanks for working on this!

LGTM

I just have one small recommended tweak.

**Review by ericsnowcurrently — CHANGES_REQUESTED** [review]
Oh, and add the missing NEWS entry.  You should use the "blurb" tool.

See: https://devguide.python.org/committing/?highlight=blurb#what-s-new-and-news-entries

**Comment by aeros167** [comment]
@ericsnowcurrently So in a case such as this where it's adding a note to the docs justifies adding news entry? 

I'm curious since I've been interested in helping with the label management on PRs with the upcoming [triager role](https://discuss.python.org/t/proposal-create-bug-triage-team-on-github/992/37) so I wanted to make sure I have a solid understanding of the appropriate usage of `skip news`. There's currently no specifications on it over on the [devguide](https://devguide.python.org/triaging/?highlight=triager).

**Comment by ericsnowcurrently** [comment]
@aeros167, we definitely add NEWS entries for documentation.  `Documentation` is dedicated section. :)

**Comment by aeros167** [comment]
>we definitely add NEWS entries for documentation. Documentation is dedicated section. :)

@ericsnowcurrently Oh no I didn't mean for documentation in general.

My initial thought was that because this PR is only adding a note to an already existing section that it would justify the usage of a `skip news` label. So my question was asking whether or not this would apply more broadly to other PRs as well. In general, it can be a little bit difficult to tell whether or not a `skip news` label is appropriate. Usually my "rule of thumb" has been that if the message conveyed is at all changed, then it should have a news entry. 

I've added news entries before when modifying sections in the docs, such as in this PR where I added the exceptions to the `os.chdir()` summary: https://github.com/python/cpython/pull/14611.

**Comment by potomak** [comment]
I have made the requested changes; please review again

**Comment by ericsnowcurrently** [comment]
@aeros167, the tricky thing is that small doc changes can have a real impact on someone out there, even an impact that you or I may not realize.  Since there isn't much cost to adding a NEWS entry, it is worth doing even for seemingly innocuous changes.  Basically the only time I use the `skip news` label is for PRs that are already covered by a NEWS entry for an earlier PR.

All that said, it would definitely be worth clarifying in the devguide.  So feel free to add a PR there (and add me as a reviewer).  You can also start a thread on the core-workflow list if you want to make sure you have the directions right.  I certainly don't speak for everyone. :)

**Review by ericsnowcurrently — APPROVED** [review]
LGTM

Thanks for doing this!

**Comment by aeros167** [comment]
@ericsnowcurrently Thanks for the clarification, I appreciate it and I can definitely see where you're coming from. Since there's no real cost associated with adding the news entry, I can understand why it would be preferable to always do so when possible. Especially if it involves anything outward facing such as the docs or a public function.  

>All that said, it would definitely be worth clarifying in the devguide. So feel free to add a PR there (and add me as a reviewer). You can also start a thread on the core-workflow list if you want to make sure you have the directions right. I certainly don't speak for everyone. :)

I've been working on helping with the new GitHub triager role documentation, currently we're still in the early stages. The next component will be adding a summary explaining the general usage for each of the labels. I figured that the `skip news` was one that was particularly important to get right, so I wanted to get clarification from the core devs as to when they think it's most appropriate. In the PR for adding in the labels section, I'll @ mention you (I don't think that I have the ability to add anyone as a reviewer as a contributor, currently only GitHub Python organization members can do that).

Thanks! (:

**Pull request merged**